colossalai/kernel/extensions/base_extension.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+from typing import Callable
+
+
+class BaseExtension(ABC):
+    @abstractmethod
+    def build_completed(self) -> bool:
```

Comment:
Just a minor suggestion, changing `build_completed` to `requires_build` might be better. The reason is that it is kind of weird to say the build is completed for triton/jit-based kernel as they do not get built at all. Using `requires_build` would make it fit with both CUDA/CPP compilation-based extension and triton/jit function based extensions.

Reply:
ok